### PR TITLE
fix: generate errors when missing lat/long for stops, stations, entrances

### PR DESCRIPTION
**Pull request by briandonahue**
**Summary:**

Generate error notice when Latitude and/or Longitude is not supplied for stops (location_type=0), stations (location_type=1) or entrances/exits (location_type=2).

Addresses [#1180](https://github.com/MobilityData/gtfs-validator/issues/1180)

**Expected behavior:** 

Should throw an error when missing geocoordinate data for stops, stations, entrances/exits

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s) (N/A)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java
Comment (bdferris-v2) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:21]: I think it'd make sense to consolidate this logic into the existing StopLatLonValidator file.

Comment (briandonahue) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:21]: @bdferris-v2 this is probably due to me being new to the project... I was only aware of the validators in the [validator](https://github.com/MobilityData/gtfs-validator/tree/master/main/src/main/java/org/mobilitydata/gtfsvalidator/validator) folder... but doing a class search, I see a generated `GtfsStopLatLonValidator` class that I assume is generated from by `LatLonValidatorGenerator`.  As you mentioned on the call earlier, it looks like maybe it already covers this scenario, as it seems to [check](https://github.com/MobilityData/gtfs-validator/blob/master/processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/LatLonValidatorGenerator.java#L96) if either coordinate is `<= 1.0` 

So now I'm wondering if I'm misunderstanding something, or if this change is unnecessary?

Comment (bdferris-v2) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:21]: Ah, duh, you are totally right, that's a generated validator :)  Sorry for the confusion, I was just doing a quick check.

Comment (bdferris-v2) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:21]: As mentioned offline, I think this class name is a little too close to the name of the existing `GtfsStopLatLonValidator` that it's probably inviting confusion (case in point, I already managed to confuse myself once here already).  I'd like to propose an alternate name.  Thoughts on one of the following?
* `StopRequiredLocationValidator`
* `StopConditionallyRequiredLocationValidator`

Basically, capturing that this validator is checking the presence of the location data itself, as opposed to validating the values in some way.  Thoughts?

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java
Comment (bdferris-v2) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:42]: I'd name this `StopWithoutLocationNotice` for consistency with the existing `StopWithoutStopTimeNotice` and `StopWithoutZoneIdNotice`

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidatorTest.java
Comment (bdferris-v2) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidatorTest.java:14]: In general, I'd prefer to inline this method and drop calls to setStopLat() and setStopLon() where stop location data is missing.  In tests, I tend to favor readability in the unit test itself over general DRY conventions for regular code.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java
Comment (briandonahue) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:21]: @bdferris-v2 Sure - I'm happy to change it.  No strong preference unless we want to get very specific (and long) with something like `LocationRequiredForStopStationEntranceValidator` or `StopStationEntranceLocationRequiredValidator` 😬Whatever feels more in line with the current naming convention...

Comment (bdferris-v2) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopLatLongValidator.java:21]: I'm not a Java long naming absolutist :)  I'd just go with `StopRequiredLocationValidator`

**Comment by briandonahue** [comment]
@bdferris-v2 I believe all your suggestions are now implemented, thanks for reviewing!

**Review by bdferris-v2 — APPROVED** [review]
LGTM

**Comment by bdferris-v2** [comment]
@briandonahue one other issue: I think the code formatting check is failing on your PR.  Can you run the Gradle `goJF` command on your code to format it properly and update the PR?  Thanks!

**Comment by isabelle-dr** [comment]
@briandonahue and @bdferris-v2 sorry to arrive late on this PR, the documentation is missing for this new notice. 
I know we are discussing generating [RULES.md](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md) automatically, but in the meantime, we need it 😊.
The HTML report links directly to RULES.md to tell users what the notice is, and how to fix it.

Also, I acknowledge that a review by a core developer is enough to unblock the merge (thank you @bdferris-v2), but please wait for a MobilityData staff to review as well before merging. I know our capacity doesn't allow us to review right away, but we need to acknowledge what has been merged on this project so that we can:
- close the issue if needed, open follow-up issues
- coordinate with other contributors (for example, PR #1315 contains the same fix)
- make sure dependencies (like the docs) have been updated
- answer questions
Many thanks 🙏

**Pull request merged**
